### ISSUE TITLE
AdyenApplePayComponent bug

### ISSUE THREAD

**Issue by drekyyy** [Adyen/adyen-flutter#269]
**Describe the bug**
Once you enter the apple pay modal, and want to close it (for example using the X button), you cant.
If you try to pay, it wont go through aswell

You can close the bottom sheet moving the app to the background but after that the button is unresponsive, you cant get the bottom sheet to pop up at all

tested on Iphone 14

Issue started appearing few days ago, after bumping the pods which was a necessary step for the adyen package to work

vid:
https://github.com/user-attachments/assets/c3015556-490c-49a9-ae38-505a4e3a2777

**Comment by Robert-SD** [comment]
Hello @drekyyy, 
thanks for reporting this behavior. It might be an iOS 18 bug and is also reported [here](https://github.com/Adyen/adyen-ios/issues/1838). 
Just to confirm, could you test older iOS versions as well? Like iOS 17? 

Bumping the pods was/is indeed necessary because we have updated the underlying iOS SDK from v5.9.0 to v5.11.0.

**Comment by kaczmarzk** [comment]
Hello @Robert-SD 

We tested this on iOS 17 and iOS 18, as well as on previous versions (v1.0.1 with the hardcoded 5.9.0 Adyen pod). 
In all cases, the close button doesn't work, there is no checkout callback, and no exceptions in Xcode.

**Comment by Robert-SD** [comment]
Hi @kaczmarzk, 
Thank you very much for testing and providing your results. We are also currently investigating the problem. We will keep you updated and as soon as a fix is available, we will release the next version.

**Comment by Robert-SD** [comment]
@kaczmarzk, just as a follow up question: Could you share the XCode version you use?

**Comment by kaczmarzk** [comment]
@Robert-SD 

[✓] Flutter (Channel stable, 3.13.9, on macOS 14.7 23H124 darwin-arm64, locale en-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)

**Comment by Robert-SD** [comment]
Thanks @kaczmarzk, this provides a different perspective.
very interesting that XCode 15 and iOS 17 also lead to the behavior. I did use this setup in the last weeks myself but will investigate further to reproduce the behavior.

**Comment by Robert-SD** [comment]
Hi @kaczmarzk and @drekyyy, 
I've adjusted the code a bit and moved the native Apple Pay component instantiation to a later stage as a proof of work. Instead of creating an instance right away when building the widget, the instance is now created when the Apple Pay button is pressed. I'm curious if this could help resolving the issue, especially if the `AdyenApplePayComponent` is building multiple times. Would it be possible to check out the branch by specifying it in your pubsepc? I want to evaluate and test the change thoroughly.

``` 
adyen_checkout:
    git:
      url: https://github.com/Adyen/adyen-flutter.git
      ref: bugfix/applePayComponentInstantiation
```

In case you want to debug the issue together, please let me know.

**Comment by Robert-SD** [comment]
Hello @kaczmarzk, @drekyyy, we released v1.1.0 containing this change. Feel free to update the version.
